### PR TITLE
fix commit view JS features, reimplement folding

### PR DESCRIPTION
**Pull request by silverwind**
File content folding was not working so I reimplemented it in a saner way. Then I noticed the issue was actually because of missing JS libraries (seen on the console of every commit with error
'SimpleMDE is not defined') resulting in most JS on that page in not being executed.

Fixed the libraries. I think the reimplementation is worth to keep.

Example page with such an error: https://try.gitea.io/silverwind/symlink-test/commit/98701cd58e49972ff22203d217176f3ad27c235c

### CODE REVIEW

File: web_src/js/index.js [hunk 1]
```diff
@@ -2109,17 +2109,12 @@ function initCodeView() {
...
-      $(e.target).parent().next().slideUp('fast', () => {
-        $foldButton.removeClass('fa-chevron-down').addClass('fa-chevron-right');
-      });
-    } else {
-      $(e.target).parent().next().slideDown('fast', () => {
-        $foldButton.removeClass('fa-chevron-right').addClass('fa-chevron-down');
-      });
-    }
+  $('.fold-code').on('click', ({ target }) => {
+    const box = target.closest('.file-content');
```
Comment (silverwind) [inline web_src/js/index.js:2114]: Note: `.closest` is not available in IE11, but I don't really care about it at this point.

Comment (silverwind) [reply web_src/js/index.js:2114]: Added a polyfill just because I'm nice.

**Comment by zeripath** [comment]
@silverwind you wouldn't be able to switch us to EasyMDE would you?

I think simplemde is a bit dead...

**Review by zeripath — APPROVED** [review]

**Comment by silverwind** [comment]
> @silverwind you wouldn't be able to switch us to EasyMDE would you?

Not sure if I'm motivated, SimpleMDE code is major spaghetti.

**Review by lunny — APPROVED** [review]

**Pull request merged**